packages/expo-image/ios/EXImageView.m

```diff
@@ -17,9 +36,67 @@ - (void)dealloc
 
 - (void)setSource:(NSDictionary *)source
 {
-  NSURL *imageURL = [RCTConvert NSURL:source[sourceUriKey]];
+  _source = source;
+  _needsReload = YES;
+}
 
-  [self sd_setImageWithURL:imageURL];
+- (void)setResizeMode:(RCTResizeMode)resizeMode
+{
+  if (_resizeMode == resizeMode) return;
+  
+  // Image needs to be reloaded whenever repeat is enabled or disabled
+  _needsReload = _needsReload || (resizeMode == RCTResizeModeRepeat) || (_resizeMode == RCTResizeModeRepeat);
+  _resizeMode = resizeMode;
+  
+  // Repeat resize mode is handled by the UIImage. Use scale to fill
+  // so the repeated image fills the UIImageView.
+  self.contentMode = resizeMode == RCTResizeModeRepeat
+    ? UIViewContentModeScaleToFill
+    : (UIViewContentMode)resizeMode;
+}
+
+- (void)didSetProps:(NSArray<NSString *> *)changedProps;
+{
+  if (_needsReload) {
+    [self reloadImage];
+  }
+}
+
+- (void)reloadImage
+{
+  _needsReload = NO;
+  
+  NSURL *imageUrl = _source ? [RCTConvert NSURL:_source[sourceUriKey]] : nil;
+  NSNumber *scale = _source && _source[sourceScaleKey] ? _source[sourceScaleKey] : nil;
+  RCTResizeMode resizeMode = _resizeMode;
+  
+  NSMutableDictionary *context = [NSMutableDictionary new];
+  
+  // Only apply custom scale factors when neccessary. The scale factor
+  // affects how the image is rendered when resizeMode `center` and `repeat`
+  // are used. On animated images, applying a scale factor may cause
+  // re-encoding of the data, which should be avoided when possible.
+  if (scale && scale.doubleValue != 1.0) {
+    [context setValue:scale forKey:SDWebImageContextImageScaleFactor];
+  }
+  
+  [self sd_setImageWithURL:imageUrl
+          placeholderImage:nil
+                   options:SDWebImageAvoidAutoSetImage
+                   context:context
+                  progress:nil
+                 completed:^(UIImage * _Nullable image, NSError * _Nullable error, SDImageCacheType cacheType, NSURL * _Nullable imageURL) {
+    
+    // Modifications to the image like changing the resizing-mode or cap-insets
+    // cannot be handled using a SDWebImage transformer, because they don't change
+    // the image-data and this causes this "meta" data to be lost in the SWWebImage caching process.
+    if (image) {
+      if (resizeMode == RCTResizeModeRepeat) {
+        image = [image resizableImageWithCapInsets:UIEdgeInsetsZero resizingMode:UIImageResizingModeTile];
+      }
+    }
+    self.image = image;
```

Comment:
Do we want to reset the `image` property even if it isn't changed when `resizeMode == repeat`? In general `sd_setImageWithURL` takes care of `self.image = image` and I may be afraid (having your comment in mind from the other PR) that setting `image` property may cause some extra operations on this `UIImageView`.

EDIT: Ok, I now saw the `SDWebImageAvoidAutoSetImage`. Do you think it would be worth to comment about it here? 😅

Reply:
This was the only way to perform the "post" processing correctly.
I feel that `SDWebImageAvoidAutoSetImage` is pretty much self explanatory